binary_search_tree/tree.py

```diff
@@ -1,59 +1,114 @@
 class TreeNode:
-    def __init__(self, key, val = None):
+    def __init__(self, key, val=None):
         if val == None:
             val = key
 
         self.key = key
         self.value = val
         self.left = None
         self.right = None
-        
-
 
 class Tree:
     def __init__(self):
         self.root = None
-
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log n)
+    # Space Complexity: O(1)
+    def add_helper(self, current, key, value):
+        if current is None:
+            return TreeNode(key, value)
+        if current.key >= key:
+            current.left = self.add_helper(current.left, key, value)
+        else:
+            current.right = self.add_helper(current.right, key, value)
+        return current
+    
     def add(self, key, value = None):
-        pass
+        if self.root == None:
+            self.root = TreeNode(key, value)
+        else:
+            self.add_helper(self.root, key, value)
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log n)
+    # Space Complexity: O(1)
     def find(self, key):
-        pass
+        current = self.root
+        while current:
+            if current.key == key:
+                return current.value
+            elif current.key > key:
+                current = current.left
+            else:
+                current= current.right
+
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
 
-    # Time Complexity: 
-    # Space Complexity: 
+    def inorder_helper(self, current, values_list):
```

Comment:
👍 , but since you're building a list of node values, the space complexity is O(n)